unishox2.h

```diff
@@ -88,19 +88,19 @@ struct us_lnk_lst {
   struct us_lnk_lst *previous;
 };
 
-extern int unishox2_compress_simple(const char *in, int len, char *out);
-extern int unishox2_decompress_simple(const char *in, int len, char *out);
-extern int unishox2_compress(const char *in, int len, char *out, 
-              const unsigned char usx_hcodes[], const unsigned char usx_hcode_lens[], 
+extern int unishox2_compress_simple(const char *in, int len, char *out, int olen);
+extern int unishox2_decompress_simple(const char *in, int len, char *out, int olen);
```

Comment:
Should we remove `olen` for at least the simple API (and pass something like 9999 as default?)  I did not add `olen` to begin with because it makes the API more difficult to use.

Reply:
ok, for lower api, we can assume `olen == -1` indicating the previous behavior, trust the developer to pass a big enough buffer. But even omit the `olen`, we should still check the result of two api not exceeds the real buffer size.

Reply:
> ok, for lower api, we can assume `olen == -1` indicating the previous behavior, trust the developer to pass a big enough buffer. But even omit the `olen`, we should still check the result of two api not exceeds the real buffer size.

The problem is C does not have optional parameters unless you use variadic functions.  So we will have to make simple api without olen and internally call with -1.